spec/graph/post_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+require 'explainer/parser'
+
+describe Post, type: :model do
+  let(:user) { create(:user) }
+  let(:posts) { FactoryGirl.create_list(:post, 5, author: user) }
+  let(:query_string) { %|
+    {
+      user(id: "#{user.to_global_id}") {
+        id
+        posts {
+          receiver {
+            id
+          }
+          author {
+            id
+          }
+        }
+      }
+    }
+  | }
+
+  it "should make N+1 queries for the authors of posts" do
+    user
+    posts
+    expect(user.posts.count).to eq 5
+    queries = ActiveRecord::Base.collecting_queries_for_explain do
+      AppSchema.execute(query_string, context: {}, variables: {})
+    end
+
+    if queries.any?
```

Comment:
You can replace this block with the following code snippet or rewrite the parser to handle explains for multiple queries.

```ruby
queries.each do |query|
  explain_output = ActiveRecord::Base.exec_explain([query])
  result = Explainer::Parser.parse(explain_output)
end
```

Reply:
I think rewriting the parser is the better option if this code snippet causes multiple roundtrips to the datastore.